AutoUpload - IllegalArgumentException: Comparison method violates its general contract!

### Actual behaviour
- AutoUpload view crashes

### Expected behaviour
- AutoUpload view shouldn't crash
 
### Steps to reproduce
1. Open the AutoUpload view in the drawer

### Environment data
Android version: 6.0.1

Device model: OnePlus 2

Stock or customized system: Stock Oxygen OS 3.6.1 

Nextcloud app version: 2.0.1

Nextcloud server version: 12.0.3

Issue reference #198

### Logs

#### Nextcloud logcat (adb logcat)
```
FATAL EXCEPTION: main
Process: com.nextcloud.client, PID: 9426
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.nextcloud.client/com.owncloud.android.ui.activity.SyncedFoldersActivity}: java.lang.IllegalArgumentException: Comparison method violates its general contract!
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2452)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2535)
	at android.app.ActivityThread.access$900(ActivityThread.java:155)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1380)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:152)
	at android.app.ActivityThread.main(ActivityThread.java:5497)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by: java.lang.IllegalArgumentException: Comparison method violates its general contract!
	at java.util.TimSort.mergeHi(TimSort.java:882)
	at java.util.TimSort.mergeAt(TimSort.java:499)
	at java.util.TimSort.mergeForceCollapse(TimSort.java:440)
	at java.util.TimSort.sort(TimSort.java:219)
	at java.util.Arrays.sort(Arrays.java:1998)
	at java.util.Collections.sort(Collections.java:1900)
	at com.owncloud.android.ui.activity.SyncedFoldersActivity.sortSyncedFolderItems(SyncedFoldersActivity.java:274)
	at com.owncloud.android.ui.activity.SyncedFoldersActivity.load(SyncedFoldersActivity.java:223)
	at com.owncloud.android.ui.activity.SyncedFoldersActivity.setupContent(SyncedFoldersActivity.java:197)
	at com.owncloud.android.ui.activity.SyncedFoldersActivity.onCreate(SyncedFoldersActivity.java:150)
	at android.app.Activity.performCreate(Activity.java:6289)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2405)
	... 9 more
```

--- a/src/main/java/com/owncloud/android/ui/activity/SyncedFoldersActivity.java
+++ b/src/main/java/com/owncloud/android/ui/activity/SyncedFoldersActivity.java
@@ -291,6 +291,8 @@ public class SyncedFoldersActivity extends FileActivity implements SyncedFolderA
                     return -1;
                 } else if (f2.getFolderName() == null) {
                     return 1;
+                } else if (PRIORITIZED_FOLDER.equals(f1.getFolderName()) && PRIORITIZED_FOLDER.equals(f2.getFolderName())) {
+                    return 0;
                 } else if (PRIORITIZED_FOLDER.equals(f1.getFolderName())) {
                     return -1;
                 } else if (PRIORITIZED_FOLDER.equals(f2.getFolderName())) {

